plugin/plugin.go

```diff
@@ -67,51 +71,66 @@ type Args struct {
 
 	// connect hostname (required)
 	ConnectHostname string `envconfig:"PLUGIN_CONNECT_HOSTNAME"`
+	// Issue Keys(optional)
+	IssueKeys []string `envconfig:"PLUGIN_ISSUEKEYS"`
 }
 
 // Exec executes the plugin.
 func Exec(ctx context.Context, args Args) error {
 	var (
-		environ  = toEnvironment(args)
-		issue    = extractIssue(args)
-		state    = toState(args)
-		version  = toVersion(args)
-		deeplink = toLink(args)
+		environ         = toEnvironment(args)
+		environmentID   = toEnvironmentId(args)
+		environmentType = toEnvironmentType(args)
+		issues          []string
+		state           = toState(args)
+		version         = toVersion(args)
+		deeplink        = toLink(args)
 	)
 
+	// ExtractInstanceName extracts the instance name from the provided URL if any
+	instanceName := ExtractInstanceName(args.Instance)
+
 	logger := logrus.
 		WithField("client_id", args.ClientID).
 		WithField("cloud_id", args.CloudID).
 		WithField("project_id", args.Project).
-		WithField("instance", args.Instance).
+		WithField("instance", instanceName).
 		WithField("pipeline", args.Name).
 		WithField("environment", environ).
 		WithField("state", state).
-		WithField("version", version)
+		WithField("environment Type", environmentType).
+		WithField("environment ID", environmentID)
 
-	if issue == "" {
-		logger.Debugln("cannot find issue number")
-		return errors.New("failed to extract issue number")
+	//check if PLUGIN_ISSUEKEYS is provided
+	if len(args.IssueKeys) > 0 {
+		issues = args.IssueKeys
+	} else {
+		// fallback to extracting from commit if no issue keys are passed
+		var issue string = extractIssue(args)
+		if issue == "" {
+			logger.Debugln("cannot find issue number")
+			return errors.New("failed to extract issue number")
+		}
+		issues = []string{issue} // add the single issue here for consistency
 	}
 
 	commitMessage := args.Commit.Message
-    	if len(commitMessage) > 255 {
-        	logger.Warnln("Commit message exceeds 255 characters; truncating to fit.")
-        	commitMessage = commitMessage[:252] + "..."
-    	}
+	if len(commitMessage) > 255 {
+		logger.Warnln("Commit message exceeds 255 characters; truncating to fit.")
+		commitMessage = commitMessage[:252] + "..."
+	}
 
-	logger = logger.WithField("issue", issue)
 	logger.Debugln("successfully extraced issue number")
-
 	deploymentPayload := DeploymentPayload{
 		Deployments: []*Deployment{
 			{
 				Deploymentsequencenumber: args.Build.Number,
 				Updatesequencenumber:     args.Build.Number,
+				IssueKeys:                issues,
```

Comment:
```suggestion
```

let's not include `IssueKeys` as it's a deprecated field as per Jira API [docs](https://developer.atlassian.com/cloud/jira/software/rest/api-group-deployments/#api-rest-deployments-0-1-bulk-post)
![Screenshot 2024-10-05 at 12 05 13 AM](https://github.com/user-attachments/assets/2603eff2-3187-424a-8e6f-aa918e922ff4)